crates/bitwarden/src/tool/generators/passphrase.rs

```diff
@@ -10,28 +13,51 @@ use serde::{Deserialize, Serialize};
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 #[cfg_attr(feature = "mobile", derive(uniffi::Record))]
 pub struct PassphraseGeneratorRequest {
+    /// Number of words in the generated passphrase.
+    /// This value must be between 3 and 20.
+    /// The default value when unset is 3
     pub num_words: Option<u8>,
+    /// Character separator between words in the generated passphrase.
+    /// If the value is set, it cannot be empty.
+    /// The default value when unset is `-`
     pub word_separator: Option<String>,
+    /// When set to true, capitalize the first letter of each word in the generated passphrase.
+    /// The default value when unset is `false`
     pub capitalize: Option<bool>,
+    /// When set to true, include a number at the end of one of the words in the generated passphrase.
+    /// The default value when unset is `false`
     pub include_number: Option<bool>,
```

Comment:
❓ I'd not use option here, since the value isn't really optional. I'd instead instantiate the type with the defaults pre-populated. If the user overrides them, then the default isn't used. That said, this is an FFI type, so I'm guessing the reason that strategy isn't used is because it doesn't map cleanly to the targeted language?

Reply:
Yeah as you mention, just having the default on the struct would be the best solution for a Rust library, but we're limited in that sense because the Default impl isn't exposed through the FFI layers at the moment.

Reply:
Ok, so I think the issue I'm running into here is that we're using the FFI type directly in the generator code. I think the solution here is to introduce a type that encapsulates the generator's requirements. The request should be mapped into that type, and the mapping process applies the defaults. I added an example to `passphrase_with_rng`.

I applied a pretty similar change in #261, but did so to implement `Distribution<char>`, which seems a bit overkill here.

Reply:
Resolved by https://github.com/bitwarden/sdk/commit/16207a6127133e70f37ec247ea79a2aaa5472959.